lib/TPP/Transforms/TileConsumerAndFuseProducers.cpp

```diff
@@ -724,7 +736,8 @@ struct TileConsumerAndFuseProducers
 
         // TODO: Remove the generalization of named ops after resolving the
         // above dependency with "populateFoldUnitExtentDimsViaSlicesPatterns".
-        linalg::populateLinalgNamedOpsGeneralizationPatterns(patterns);
+        if (!this->keepNamedOp)
+          linalg::populateLinalgNamedOpsGeneralizationPatterns(patterns);
```

Comment:
I'm not against a flag for control per se but I fail to see how this addresses or impacts anything.
Is it a setup for some other long-term changes? Could this be added to some other PR that utilizes this change?

Reply:
Yes, I was thinking to add more test case where we could see the impact of tile-and-fuse in named ops. 

I can take it up in a separate PR too. In fact, I think this generalization is redundant now as no test was failing but not sure if this is the case also with CI and benchmarks. Let me know if you have any info.

Reply:
I don't think generalization destroys the test case you're looking for i.e. `linalg.add ins(%A, %B) outs(%A)` generalizes to `linalg.generic ins(%A, %B) outs(%A)`. So, let's remove this change to the tile-and-fuse pass from this PR.

If you want to, let's try removing generalization in a separate PR. Then we can focus on impact on benchmarks etc.